src/main/java/server/translator/Translator.java

```diff
@@ -0,0 +1,279 @@
+package server.translator;
+
+import commons.requests.BookFacilityRequest;
+import commons.requests.*;
+import commons.responses.*;
+import commons.rpc.ServerCommunicator;
+import commons.utils.Datetime;
+import commons.utils.ResponseMessage;
+import javafx.util.Pair;
+import server.entities.Booking;
+import server.entities.FacilityObserver;
+import server.managers.BookingManager;
+import server.managers.ObserverManager;
+
+import java.net.InetAddress;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Hashtable;
+import java.util.List;
+
+public class Translator {
+    private ArrayList<String> allDays = new ArrayList<String>();
+    private ServerCommunicator serverCommunicator;
+
+    public Translator(ServerCommunicator serverCommunicator){
+        List<String> days = Arrays.asList("Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday");
+        allDays.addAll(days);
+
+        this.serverCommunicator = serverCommunicator;
+    }
+
+    public QueryAvailabilityResponse queryAvailability(QueryAvailabilityRequest r, BookingManager bookingManager, Hashtable facilTable) {
+
+        QueryAvailabilityResponse response;
+        ResponseMessage responseMessage;
+
+        String facilName = r.facilityName;
+        ArrayList days = (ArrayList) r.days;
+
+        Pair<Hashtable, Exception> outputPair = bookingManager.queryAvailability(facilName, days, facilTable);
+        Hashtable outputBooking = outputPair.getKey();
+        Exception outputException = outputPair.getValue();
+
+        if (outputException != null){
+            responseMessage = new ResponseMessage(400, outputException.getMessage()) ;
+            response = new QueryAvailabilityResponse(facilName, null, responseMessage);
+        }
+        else{
+            responseMessage = new ResponseMessage(200, "success") ;
+            ArrayList convertedList =localToDatetime(outputBooking);
```

Comment:
Serializer has java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0 error for this list